### PR TITLE
[MM-56491] Allow sysadmins to LDAP sync SAML users when SamlSettings.EnableSyncWithLdap is true

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Allowing sysadmins to ldap sync saml users when SamlSettings.EnableSyncWithLdap is true

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56491

#### Release Note

```release-note
Allow sysadmins to LDAP sync SAML users when SamlSettings.EnableSyncWithLdap is true
```

### CODE REVIEW

**Review by JulienTant — APPROVED** [review]
Nice change!

File: webapp/channels/src/components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx [hunk 1]
```diff
@@ -737,7 +737,7 @@ export default class SystemUsersDropdown extends React.PureComponent<Props, Stat
                         </SystemPermissionGate>
                         <SystemPermissionGate permissions={[Permissions.SYSCONSOLE_WRITE_USERMANAGEMENT_GROUPS]}>
                             <Menu.ItemAction
-                                show={user.auth_service === Constants.LDAP_SERVICE}
+                                show={user.auth_service === Constants.LDAP_SERVICE || (user.auth_service === Constants.SAML_SERVICE && config.SamlSettings?.EnableSyncWithLdap)}
```
Comment (hanzei) [inline webapp/channels/src/components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx:740]: If LDAP sync is enabled, will this show for non-LDAP user, which authenticate using SAML?

Is there a way to check where a user originated from, i.e. if it's created via Email/Password or by the LDAP server?

File: server/channels/api4/ldap_test.go [hunk 2]
```diff
@@ -309,4 +309,26 @@ func TestAddUserToGroupSyncables(t *testing.T) {
...
+	require.Error(t, err)
+	CheckForbiddenStatus(t, resp)
+
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.SamlSettings.EnableSyncWithLdap = true
+	})
+
+	resp, err = th.SystemAdminClient.AddUserToGroupSyncables(context.Background(), user.Id)
+	require.NoError(t, err)
+	CheckOKStatus(t, resp)
```
Comment (hanzei) [inline server/channels/api4/ldap_test.go:333]: Nit: Can we move this into a subtest to make it clear what the test tests for?

**Review by hanzei — CHANGES_REQUESTED** [review]
Nice QOL improvement :+1:

File: webapp/channels/src/components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx (hunk 1 above)
Comment (BenCookie95) [reply webapp/channels/src/components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx:740]: Unfortunately I don't think we have a way to distinguish this. Is it even possible to have non-LDAP SAML users if that sync is enabled?

File: server/channels/api4/ldap_test.go [hunk 3]
```diff
@@ -309,4 +309,28 @@ func TestAddUserToGroupSyncables(t *testing.T) {
 	resp, err = th.SystemAdminClient.AddUserToGroupSyncables(context.Background(), user.Id)
 	require.NoError(t, err)
 	CheckOKStatus(t, resp)
+
+	t.Run("should sync SAML users when SamlSettings.EnableSyncWithLdap is true", func(t *testing.T) {
```
Comment (hanzei) [inline server/channels/api4/ldap_test.go:313]: Nit-pick: Nice to see that you introduce a subtest. Maybe we also convert the other to a sub test while we are here?

File: webapp/channels/src/components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx (hunk 1 above)
Comment (hanzei) [reply webapp/channels/src/components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx:741]: You are correct. All SAML users will have their attributes synced with LDAP if `EnableSyncWithLdap` is enabled.

**Review by hanzei — APPROVED** [review]
One nit pick, otherwise LGTM :rocket:

**Comment by hanzei** [comment]
@BenCookie95 Heads up that there is a conflict to resolve.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- I have tested to ensure the new button works to run the sync for the individual user
- I can see in the logs the call goes out to `group_sync_memberships` for the user ID

**note** Because I don't have a setup where the same identity provider is used for both SMAL and LDAP so I wan't able to to see the sync manipulate group data. But, I do see a sync triggering.

LGTM!

**Review by JulienTant — APPROVED** [review]

**Comment by hanzei** [comment]
Merging as @BenCookie95 is OOO.

**Pull request merged**